modules/light-clients/08-wasm/types/misbehaviour_handle_test.go

```diff
@@ -469,372 +481,128 @@ package types_test
+					resp, err := json.Marshal(types.CheckForMisbehaviourResult{FoundMisbehaviour: true})
+					suite.Assert().NoError(err)
+					return resp, types.DefaultGasUsed, nil
+				})
 			},
 			true,
+			false,
 		},
 		{
-			"invalid wasm misbehaviour",
-			func() {
-				clientMsg = &solomachine.Misbehaviour{}
+			"contract error, resp cannot be marshalled", func() {
+				suite.mockVM.RegisterQueryCallback(types.CheckForMisbehaviourMsg{}, func(codeID wasmvm.Checksum, env wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					resp := "cannot be unmarshalled"
+					return []byte(resp), types.DefaultGasUsed, nil
+				})
+			},
+			false,
+			false,
+		},
+		{
+			"vm returns error, ", func() {
+				suite.mockVM.RegisterQueryCallback(types.CheckForMisbehaviourMsg{}, func(codeID wasmvm.Checksum, env wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					return nil, 0, errors.New("invalid block ID")
+				})
 			},
 			false,
+			false,
+		},
+		{
+			"contract panics, panic propogated", func() {
+				panicErr = errors.New("panic in query to contract")
+				suite.mockVM.RegisterQueryCallback(types.CheckForMisbehaviourMsg{}, func(codeID wasmvm.Checksum, env wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					panic(panicErr)
+				})
+			},
+			false,
+			true,
 		},
 	}
 
 	for _, tc := range testCases {
 		suite.Run(tc.name, func() {
-			suite.SetupWasmGrandpaWithChannel()
-			clientState, ok = suite.chainA.App.GetIBCKeeper().ClientKeeper.GetClientState(suite.ctx, grandpaClientID)
-			suite.Require().True(ok)
+			// reset suite to create fresh application state
+			suite.SetupWasmWithMockVM()
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+			err := endpoint.CreateClient()
+			suite.Require().NoError(err)
+
+			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), endpoint.ClientID)
+			clientState := endpoint.GetClientState()
 
 			tc.malleate()
 
-			foundMisbehaviour := clientState.CheckForMisbehaviour(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
+			data, err := base64.StdEncoding.DecodeString(suite.testData["header"])
+			suite.Require().NoError(err)
+			clientMessage := &types.ClientMessage{
+				Data: data,
+			}
 
-			if tc.expPass {
-				suite.Require().True(foundMisbehaviour)
+			if !tc.expPanic {
+				foundMisbehaviour = clientState.CheckForMisbehaviour(suite.chainA.GetContext(), suite.chainA.App.AppCodec(), clientStore, clientMessage)
```

Comment:
```suggestion
				foundMisbehaviour = clientState.CheckForMisbehaviour(suite.ctx, suite.chainA.App.AppCodec(), suite.store, clientMessage)
```
You can drop clientStore and `GetContext` entirely. [This was added in a recent commit.](https://github.com/cosmos/ibc-go/blob/feat/wasm-clients/modules/light-clients/08-wasm/types/types_test.go#L84-L85)